tests/components/smartthings/test_climate.py

```diff
@@ -465,6 +465,62 @@ async def test_ac_set_preset_mode(
     )
 
 
+@pytest.mark.parametrize("device_fixture", ["da_ac_rac_000002"])
+async def test_ac_set_and_get_preset_mode(
+    hass: HomeAssistant,
+    devices: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test setting and retrieving AC preset modes."""
+    await setup_integration(hass, mock_config_entry)
+
+    # Mock supported preset modes
+    set_attribute_value(
+        devices,
+        Capability.CUSTOM_AIR_CONDITIONER_OPTIONAL_MODE,
+        Attribute.SUPPORTED_AC_OPTIONAL_MODE,
+        ["off", "sleep", "quiet", "speed", "windFree", "windFreeSleep"],
+    )
+
+    # Test setting each preset mode
+    for mode in ("off", "sleep", "quiet", "speed", "windFree", "windFreeSleep"):
+        await hass.services.async_call(
+            CLIMATE_DOMAIN,
+            SERVICE_SET_PRESET_MODE,
+            {ATTR_ENTITY_ID: "climate.ac_office_granit", ATTR_PRESET_MODE: mode},
+            blocking=True,
+        )
+        devices.execute_device_command.assert_called_with(
+            "13549124-3320-4fda-8e5c-3f363e043034",
+            Capability.CUSTOM_AIR_CONDITIONER_OPTIONAL_MODE,
+            Command.SET_AC_OPTIONAL_MODE,
+            MAIN,
+            argument=mode,
+        )
+
+        # Mock the current preset mode to simulate the device state
+        set_attribute_value(
```

Comment:
can we move this to a separate test? I.e. one for setting and another for getting the mode. That will ensure that the service call is not what is changing the state.